tests/test_provider.py

```diff
@@ -36,6 +40,10 @@ def test_connect_and_disconnect(disconnected_provider):
     assert not disconnected_provider.is_connected
     assert disconnected_provider.process is None
 
+    # Proof it is really disconnected.
```

Comment:
this would not raise any exceptions on `main` branch (i made sure). The reason is because the node never actually disconnects. Now it does, and this fails expectedly. Thus, we have a test! yay